detox/webhook_server.js

```diff
@@ -0,0 +1,296 @@
+        sendSysadminResponse(message, body.channel_id);
+    } else {
+        message = 'Dialog submitted';
+        sendSysadminResponse(message, body.channel_id);
+    }
+
+    return res.json({text: message});
+}
+
+/**
+ * @route "POST /send_message_to_channel?type={messageType}&channel_id={channelId}"
+ * @query type - message type of empty string for regular message if not provided (default), "system_message", etc
+ * @query channel_id - channel where to send the message
+ */
+function postSendMessageToChannel(req, res) {
+    const channelId = req.query.channel_id;
+    const response = {
+        response_type: 'in_channel',
+        text: 'Extra response 2',
+        channel_id: channelId,
+        extra_responses: [{
+            response_type: 'in_channel',
+            text: 'Hello World',
+            channel_id: channelId,
+        }],
+    };
+
+    if (req.query.type) {
+        response.type = req.query.type;
+    }
+
+    res.json(response);
+}
+
+function getWebhookBaseUrl() {
+    return WEBHOOK_BASE_URL || 'http://localhost:3000';
+}
+
+function getBaseUrl() {
+    return SITE_URL || 'http://localhost:8065';
+}
+
+// Convenient way to send response in a channel by using sysadmin account
+function sendSysadminResponse(message, channelId) {
+    const username = ADMIN_USERNAME || 'sysadmin';
+    const password = ADMIN_PASSWORD || 'Sys@dmin-sample1';
+    const baseUrl = getBaseUrl();
+    postMessageAs({sender: {username, password}, message, channelId, baseUrl});
+}
+
+const responseTypes = ['in_channel', 'comment'];
+
+function getWebhookResponse(body, {responseType, username, iconUrl}) {
+    const payload = Object.entries(body).map(([key, value]) => `- ${key}: "${value}"`).join('\n');
+
+    return `
+\`\`\`
+#### Outgoing Webhook Payload
+${payload}
+#### Webhook override to Mattermost instance
+- response_type: "${responseType}"
+- type: ""
+- username: "${username}"
+- icon_url: "${iconUrl}"
+\`\`\`
+`;
+}
+
+/**
+ * @route "POST /post_outgoing_webhook?override_username={username}&override_icon_url={iconUrl}&response_type={comment}"
+ * @query override_username - the user name that overrides the user name defined by the outgoing webhook
+ * @query override_icon_url - the user icon url that overrides the user icon url defined by the outgoing webhook
+ * @query response_type - "in_channel" (default) or "comment"
+ */
+function postOutgoingWebhook(req, res) {
+    const {body, query} = req;
+    if (!body) {
+        res.status(404).send({error: 'Invalid data'});
+    }
+
+    const responseType = query.response_type || responseTypes[0];
+    const username = query.override_username || '';
+    const iconUrl = query.override_icon_url || '';
+
+    const response = {
+        text: getWebhookResponse(body, {responseType, username, iconUrl}),
+        username,
+        icon_url: iconUrl,
+        type: '',
+        response_type: responseType,
+    };
+    res.status(200).send(response);
+}
```

Comment:
Looks good to me but maybe in the future, let's extract this into its own repo/package so it can be shared not only between webapp and mobile, but other repos too, like plugins.

Reply:
@saturninoabril Yeah, we can definitely extract this in the future.